tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs

```diff
@@ -848,9 +848,13 @@ public void TestSocketConfigurator()
             settings.ClusterConfigurator = cb => cb.ConfigureTcp(tcp => tcp.With(socketConfigurator: socketConfigurator));
             var subject = new MongoClient(settings);
 
-            SpinWait.SpinUntil(() => subject.Cluster.Description.State == ClusterState.Connected, TimeSpan.FromSeconds(5)).Should().BeTrue();
+            var isConnected = SpinWait.SpinUntil(() => subject.Cluster.Description.State == ClusterState.Connected, TimeSpan.FromSeconds(5));
+            if (!isConnected)
+            {
+                throw new Exception($"The cluster connecting exceeded the timeout 5sec. ClusterDescription is {subject.Cluster.Description}.");
+            }
```

Comment:
I'm surprised you felt this change was necessary.

It causes no harm, but we don't expect this call to `SpinUntil` to return false, and in the rare occasion that it does we don't necessarily need a custom error message. It would be sufficient for `Should().BeTrue()` put out its standard message.

Note that we have lots of places where we call `SpinUntil` followed by `Should().BeTrue()` where we don't bother with custom error messages. Not sure it's worth it or that we want to start doing that.

Reply:
yes, we usually don't do it, but I faced with this error during testing (when 5 sec wasn't enough). So i would add this to have additional info just in case.

Reply:
It's unnecessary but I won't stop you. I feel no need to change any other calls to `SpinUntil`.

Reply:
I agree that it doesn't make a lot of sense to use it everywhere. But the previous error is really useless if you can't reproduce the failed situation locally.